playwright-e2e/dss/component/table.ts

```diff
@@ -262,18 +261,83 @@ export default class Table {
     }
   }
 
-  async searchByColumn(column1Name: string, value1: string, opts?: { column2Name: string, value2: string }): Promise<void> {
+  async searchByColumn(column1Name: string, value1: string, opts: { column2Name?: string, value2?: string, clear?: boolean } = {}): Promise<void> {
+    const { column2Name, value2, clear = true } = opts;
     const column1Index = await this.getHeaderIndex(column1Name, { exactMatch: false });
     const input1 = this.page.locator(this.headerRowCss).nth(1).locator('th').nth(column1Index).locator('input.form-control');
+    if (clear) {
+      await input1.clear();
+    }
     await input1.fill(value1);
-    if (opts) {
-      const column2Index = await this.getHeaderIndex(opts.column2Name, { exactMatch: false });
+    if (column2Name && value2) {
+      const column2Index = await this.getHeaderIndex(column2Name, { exactMatch: false });
       const input2 = this.page.locator(this.headerRowCss).nth(1).locator('th').nth(column2Index).locator('input.form-control');
-      await input2.fill(opts.value2);
+      await input2.fill(value2);
     }
     await waitForNoSpinner(this.page);
   }
 
+  /**
+   * Click "Select" checkbox by column and cell text.
+   * @param columnHeader {string} Search column name.
+   * @param columnCellText {string} Cell text.
+   * @param opts: Optional flag for text exact match
+   * @returns
+   */
+  async selectRowByColumn(columnHeader: string, columnCellText: string, opts: { exactMatch?: boolean } = {}): Promise<Checkbox> {
+    const { exactMatch = true } = opts;
+
+    // Find column header index
+    const columnIndex = await this.getHeaderIndex(columnHeader, { exactMatch });
+    if (columnIndex === -1) {
+      throw new Error(`Column: ${columnHeader} not found.`);
+    }
+
+    // Find row which contains searchCellValue
+    const allRows = await this.rowLocator().elementHandles();
+    const allCellValues = await Promise.all(
+      allRows.map(async (row) => {
+        const cells = await row.$$(this.cellCss);
+        return cells[columnIndex].innerText();
+      })
+    );
+    const searchRowIndex = allCellValues.findIndex((cellValue) => cellValue === columnCellText);
+    if (searchRowIndex === -1) {
+      throw new Error(`Column cell text: ${columnCellText} not found.`);
+    }
+
+    const cell = this.cell(searchRowIndex, 0); // column index is 0 because Select checkbox is located on first column
+    const checkbox = new Checkbox(this.page, {root: cell});
+    await checkbox.click();
+    return checkbox;
+  }
+
+  async selectSingleRowByIndex(rowIndex = 0): Promise<void> {
+    const cell = this.cell(rowIndex, 0);
+    const checkbox = new Checkbox(this.page, { root: cell });
+    await checkbox.click();
+    await waitForNoSpinner(this.page);
+  }
+
+  public async getTextAt(rowIndex: number, columnName: string, opts: { exactMatch?: boolean } = {}): Promise<string[]> {
```

Comment:
Generic version of function [`getParticipantDataAt`](https://github.com/broadinstitute/ddp-angular/blob/7b41d2257327ae35105a6064658aafa31ddbd4af/playwright-e2e/dsm/component/tables/participant-list-table.ts#L46)